### ISSUE TITLE
Better handling of bad flow

### ISSUE THREAD

**Issue by corentinghigny** [kestra-io/kestra#98]
A bad flow creation should not return a 5XX, it should return a 422 or something like that

I'll split them one message by error so we can discuss them independently, and add others if I make some discoveries

 - **Input**
```yaml
```

(empty)

- **Result**

`Internal server error: Required Body [flow] not specified`

**Comment by corentinghigny** [comment]
- **Input**
```yaml
id: toto
```

- **Result**

`Illegal argument: value cannot be null`

This "argument" is the namespace but this is just an educated guess, it is not explicit and it should be.

**Comment by corentinghigny** [comment]
-------------------------------------------------------------
 - **Input**
```yaml
id: toto
namespace: tata
```

- **Result**

:warning: **Flow is created**

`Internal server error: null`

The flow is created, I get redirected to its page with source code :
```yaml
id: toto
namespace: titi
revision: 1
```

The `null` error seems link to the fact that we have no task.

In the server terminal :
```
2020-03-20 11:33:17,669 ERROR oopGroup-1-6 orController null
java.lang.NullPointerException: null
        at org.kestra.core.services.TreeService.sequential(TreeService.java:87)
        at org.kestra.core.services.TreeService.sequential(TreeService.java:67)
        at org.kestra.core.services.TreeService.sequential(TreeService.java:24)
        at org.kestra.core.models.hierarchies.FlowTree.of(FlowTree.java:23)
        at org.kestra.core.models.hierarchies.FlowTree.of(FlowTree.java:18)
```

**Comment by corentinghigny** [comment]
- **Inputs**
```yaml
id: toto
namespace: tutu
inputs:
 - name: wrong
 - type: foobar
```

- **Result**
```
Internal server error: Failed to convert argument [flow] for value [null] due to: 
Cannot deserialize value of type `org.kestra.core.models.flows.Input$Type` from String "foobar": 
not one of the values accepted for Enum class: 
[STRING, DATETIME, FLOAT, INT, FILE]
at [Source: UNKNOWN; line: -1, column: -1] 
(through reference chain: org.kestra.core.models.flows.Flow["inputs"]
->java.util.ArrayList[1]
->org.kestra.core.models.flows.Input["type"])
```

So I can "see" that I have not a valid input type, but that is a pretty scary error message. And an internal error, so 5XX :cry:  Just put a little makeup on it!

**Comment by corentinghigny** [comment]
- **Inputs**
```yaml
id: toto
namespace: tata
inputs:
 - name: wrong
```

- **Result**

:warning: **Flow is created*
```yaml
id: toto
namespace: tata
revision: 1
inputs:
  - name: input
    required: true
```

It should not be created, shouldn't it fail with a error message : `type should be one of [STRING, DATETIME, FLOAT, INT, FILE]` or something like that ? Type shouldn't be optional (or should it ?)

**Comment by corentinghigny** [comment]
- **Inputs**
```yaml

id: titi
namespace: toto
tasks:
  - id: task
    type: fr.leroymerlin.bad.type
    format: hello
```

 - **Result**

```
Internal server error: Failed to convert argument [flow] for value [null] due to: 
Could not resolve type id 'fr.leroymerlin.bad.type' as a subtype of `org.kestra.core.models.tasks.Task`: 
no such class found at [Source: UNKNOWN; line: -1, column: -1] 
(through reference chain: org.kestra.core.models.flows.Flow["tasks"]->java.util.ArrayList[0])
io.micronaut.core.convert.exceptions.ConversionErrorException: 
Failed to convert argument [flow] for value [null] due to: Could not resolve type id 
'fr.leroymerlin.bad.type' as a subtype of `org.kestra.core.models.tasks.Task`: no such class found
```
Expected a better error message

**Comment by eregnier** [comment]
statement for these issues :
* 1 : solved in 6860c43
* 2 : solved in 6860c43
* 3 : The issues comes from the tree route that does not manage  an empty flow. I guess we should prevent empty flow creation as this is "noise". this have to be done serverside and handled in frontend.
* 4 : solved in 6860c43
* 5 : same as  3, tree route does not handle such erroneous flow. We should control this at creation time, or maybe this is just the tree route that does not handle this properly.
* 6 : hum, yes. at least 6860c43 displays it properly.
